### ISSUE TITLE
`auto_resize` doesn't work

### ISSUE THREAD

**Issue by LPalmito** [BTMorton/angular2-grid#52]
Hey!

I would like my grid items to resize when needed in order to fit the 6 columns I've created, just like in this example: https://bmorton.co.uk/angular/. But when I put `max_cols` to something else than 0 and `auto_resize` to true (in the gridConfig), my browser is loading indefinitely the webpage instead of displaying what it should.

If anyone could help that would be really nice , thanks! :)

**Comment by BTMorton** [comment]
Do you have an online example anywhere that I could look at or the error output at all?

**Comment by LPalmito** [comment]
Yes, sorry! I think I have managed to locate the issue more precisely for you: it happens when I try to display grid items of different sizes. Here is a sample of my code:

```
<div *ngFor='#widget of widgets; #i = index'>
                    <div [ngGridItem]="{'sizex': '2', 'sizey': '3'}" *ngIf="widget.permanentInfo.type == 'graph'">
                        <graph-widget [currentWidget]="widget"></graph-widget>
                    </div>
                    <div [ngGridItem] *ngIf="widget.permanentInfo.type == 'text'">
                        <text-widget [currentWidget]="widget"></text-widget>
                    </div>
</div>
```

When I don't specify `sizex` and `sizey` to something different, there is no problem. I hope it'll help you! :)

**Comment by LPalmito** [comment]
OK, even if I don't really understand why, I don't have this issue anymore if I do that:
```
<div *ngFor='#widget of widgets; #i = index' [ngGridItem]="getGridInfo(widget)">
                    <div *ngIf="widget.permanentInfo.type == 'graph'">
                        <graph-widget [currentWidget]="widget"></graph-widget>
                    </div>
                    <div *ngIf="widget.permanentInfo.type == 'text'">
                        <text-widget [currentWidget]="widget"></text-widget>
                    </div>
</div>
```
where `getGridInfo(widget)` is a method in my component which returns me the right `col`, `row`, `sizex`, `sizey` according to the widget and the size of the window.

Everything works perfectly now, the display of the widgets and the rescaling of the window, which mimic kind of a responsive feature, except when rescaling from `md` window to `sm` window, which makes all crash and nothing is working anymore.

If you want more information about the methods I'm using in my component:

- to test the size of the window (return true if `alias` is the size of the window (`xs`, `sm`, `md` or `lg`):
```
    isBreakpoint(alias) {
        return $('.test-' + alias).is(':visible');
    }
```

- to adapt the total number of columns according to the size of the window:
```
    @HostListener('window:resize')
    loadGrid() {
        if (this.isBreakpoint('sm') || this.isBreakpoint('xs')) {this.gridConfig.max_cols = 1;}
        else if (this.isBreakpoint('md')) {this.gridConfig.max_cols = 2;}
        else if (this.isBreakpoint('lg')) {this.gridConfig.max_cols = 3;}
    }
```

- to get the right information concerning `col`, `row`, `sizex`, `sizey` of each widget according to the size of the window:
```
    getGridInfo(widget) {
        if (this.isBreakpoint('sm') || this.isBreakpoint('xs')) {return widget.gridInfo.sm}
        else if (this.isBreakpoint('md')) {return widget.gridInfo.md}
        else if (this.isBreakpoint('lg')) {return widget.gridInfo.lg}
    }
```

Hope it would be useful for you!

**Comment by BTMorton** [comment]
Got it! It's the fact that you did {'size_x': '2', 'size_y': '3'}, it was treating them as strings, not numbers and breaking everything. I'll make a fix and parseInt everything haha

**Comment by BTMorton** [comment]
Fixed by 888f82903858c6c9dd7cba010380897243db253a

**Comment by LPalmito** [comment]
Thanks a lot! The situation I was describing in my previous comment was in fact coming from an other issue, I have made a new post for it :)
